meta.nabl2.shared/syntax/nabl2/shared/constraints/Namebinding.sdf3

```diff
@@ -16,6 +16,9 @@ context-free syntax
   Constraint.CGDecl       = [[Occurrence] <- [Scope] [MessagePosition]]
   Constraint.CGRef        = [[Occurrence] -> [Scope] [MessagePosition]]
 
+  Constraint.CFDecl       = [[Occurrence] <~- [CFGNode] [MessagePosition]]
+  Constraint.CFDirectEdge = [[CFGNode] -~-> [CFGNode] [MessagePosition]]
```

Comment:
When are these productions used? Do you expect people to include these constraints in NaBL2 specs, or is it just for pretty printing?

Reply:
It's just for prettyprinting